gateway/ln-gateway/src/rpc/rpc_server.rs

```diff
@@ -83,23 +85,60 @@ fn extract_bearer_token(request: &Request) -> Result<String, StatusCode> {
 }
 
 /// Middleware to authenticate an incoming request. Routes that are
-/// authenticated with this middleware always require a Bearer token to be
-/// supplied in the Authorization header.
-async fn auth_middleware(
+/// authenticated with this middleware always require a Bearer token
+/// with the JWT generated previously to be supplied in the Authorization
+/// header. If jwt fails, try to check authentication with password
+async fn auth_jwt_middleware_with_password_fallback(
     Extension(gateway): Extension<Arc<Gateway>>,
     request: Request,
     next: Next,
 ) -> Result<impl IntoResponse, StatusCode> {
     let token = extract_bearer_token(&request)?;
+    let auth_manager = gateway.auth_manager.lock().await;
+    if let Ok(decoded_token_data) = jsonwebtoken::decode::<crate::auth_manager::Session>(
+        &token,
+        &jsonwebtoken::DecodingKey::from_secret(auth_manager.encoding_secret.as_ref()),
+        &jsonwebtoken::Validation::default(),
+    ) {
+        let now = fedimint_core::time::now()
+            .duration_since(UNIX_EPOCH)
+            .expect("Time went backwards")
+            .as_secs();
+        if now > decoded_token_data.claims.exp {
+            return Err(StatusCode::UNAUTHORIZED);
+        }
```

Comment:
This could be a separate function for checking the JWT, would make the code cleaner. 

EDIT: should live inside `AuthManager` after reading the full diff